tracing-opentelemetry/src/tracer.rs

```diff
@@ -56,14 +56,18 @@ impl PreSampledTracer for api::NoopTracer {
 
 impl PreSampledTracer for sdk::Tracer {
     fn sampled_span_context(&self, builder: &mut api::SpanBuilder) -> api::SpanContext {
-        let span_id = builder.span_id.expect("Builders must have id");
+        let span_id = builder
+            .span_id
+            .unwrap_or_else(|| self.provider().config().id_generator.new_span_id());
```

Comment:
TIOLI: it seems like this could _potentially_ be factored out into a function, but it may not really be worth it since it's only used twice.

Reply:
Yeah think this whole method could use a refactor. Will do in future PR 👍